packages/rabbitmq/src/rabbitmqTxSubmitProvider.ts

```diff
@@ -0,0 +1,112 @@
+import { Buffer } from 'buffer';
+import { Cardano, HealthCheckResponse, ProviderError, ProviderFailure, TxSubmitProvider } from '@cardano-sdk/core';
+import { Channel, Connection, connect } from 'amqplib';
+import { Logger, dummyLogger } from 'ts-log';
+
+const queue = 'tx-submission';
+
+/**
+ * Connect to a [RabbitMQ](https://www.rabbitmq.com/) instance
+ */
+export class RabbitMqTxSubmitProvider implements TxSubmitProvider {
+  #channel?: Channel;
+  #connection?: Connection;
+  #connectionURL: URL;
+  #logger: Logger;
+  #queueWasCreated = false;
+
+  /**
+   * @param {URL} connectionURL RabbitMQ connection URL
+   * @param {Logger} logger object implementing the Logger abstract class
+   */
+  constructor(connectionURL: URL, logger: Logger = dummyLogger) {
+    this.#connectionURL = connectionURL;
+    this.#logger = logger;
+  }
+
+  /**
+   * Connects to the RabbitMQ server and create the channel
+   */
+  async #connectAndCreateChannel() {
+    if (this.#connection) return;
+
+    try {
+      this.#connection = await connect(this.#connectionURL.toString());
+    } catch (error) {
+      await this.close();
+      throw new ProviderError(ProviderFailure.ConnectionFailure, error);
+    }
+
+    try {
+      this.#channel = await this.#connection.createChannel();
+    } catch (error) {
+      await this.close();
+      throw new ProviderError(ProviderFailure.ConnectionFailure, error);
+    }
+  }
+
+  /**
+   * Idempotently (channel.assertQueue does the job for us) creates the queue
+   *
+   * @param {boolean} force Forces the creation of the queue just to have a response from the server
+   */
+  async #createQueue(force?: boolean) {
```

Comment:
```suggestion
  async #ensureQueue(force?: boolean) {
```
This communicates the behaviour a little better and improves:

``` TypeScript
async submitTx(signedTransaction: Uint8Array) {
    try {
      await this.#ensureQueue();
      this.#channel!.sendToQueue(queue, Buffer.from(signedTransaction));
      ...
```